File: librarySystem/Periodical.cpp
#include <iostream>
#include "Periodical.h"
#include "Periodical.h"
#include <sstream>

using namespace std;

Periodical::Periodical(string &cno, string &titl, string &sub, string &auth, string &desc, string &publ, string &publishingHist
    , string &ser, string &note, string &relatedTit, string &otherTitles, string &govDocNum) :
    Media(cno, titl, sub, note), author(auth), description(desc), publisher(publ), publishingHistory(publishingHist),
    series(ser), relatedTitles(relatedTit), otherFormsOfTitle(otherTitles), govtDocNumber(govDocNum)
{
}

Periodical::~Periodical()
{
}

void Periodical::display() const
{
	cout<<"----------------------Periodical Details-------------------------"<<endl;
	Media::display();
	cout<<"Author-----------------"<<this->author<<endl;
	cout<<"Description------------"<<this->description<<endl;
	cout<<"Publisher--------------"<<this->publisher<<endl;
	cout<<"Publishing History-----"<<this->publishingHistory<<endl;
	cout<<"Series-----------------"<<this->series<<endl;
	cout<<"Notes------------------"<<this->notes<<endl;
	cout<<"Related Titles---------"<<this->relatedTitles<<endl;
	cout<<"Other Forms of Title---"<<this->otherFormsOfTitle<<endl;
	cout<<"Govt. Doc Number-------"<<this->govtDocNumber<<endl;
	cout<<"-----------------------------------------------------------------"<<endl;
}

bool Periodical::compare_other(const string &other) const
{
	if((this->description.find(other) != std::string::npos) ||
		(this->notes.find(other) != std::string::npos) ||
		(this->series.find(other) != std::string::npos) ||
		(this->relatedTitles.find(other) != std::string::npos))
	{
		return true;
	}
	else
		return false;
}

File: libPrjPython/Periodical.py
from Media import Media

class Periodical(Media):
    
    def __init__(self, callno, title, subject, author, description, publisher, publishing_history, series, notes, related_titles,
                 other_forms_of_title, govt_doc_no):
        super(Periodical, self).__init__(callno, title, subject, notes)
        self.author = author
        self.description = description
        self.publisher = publisher
        self.publishing_history = publishing_history
        self.series = series
        self.related_titles = related_titles
        self.other_forms_of_title = other_forms_of_title
        self.govt_doc_no = govt_doc_no
    
    def display(self):
       print "---------------Periodical Details-------------------------------------------------------------------------"
       super(Periodical, self).display()
       print "Author               : ", self.author
       print "Description          : ", self.description
       print "Publisher            : ", self.publisher
       print "Publishing History   : ", self.publishing_history
       print "Series               : ", self.series
       print "Notes                : ", self.notes
       print "Related Titles       : ", self.related_titles
       print "Other Forms of Title : ", self.other_forms_of_title
       print "Govt. Doc No.        : ", self.govt_doc_no
       print "\n"

    def compare_other(self,srch_other):
        return srch_other in self.description or srch_other in self.notes or srch_other in self.series                                        or srch_other in self.related_titles 
